### PR TITLE
Added search logic

### PR DESCRIPTION
**Pull request by vitalii-bashta**

### CODE REVIEW

File: src/app/home/services/events.service.ts [hunk 1]
```diff
@@ -8,30 +8,22 @@ import { HttpServiceEvents } from '../../core/services/http-events.service';
...
+	constructor(private httpServiceEvents: HttpServiceEvents) { }
 
-    getDataUpdated() {
-        return this.dataUpdated.asObservable().pipe(
-            startWith(''),
-            debounceTime(1000)
-        );
-    }
+	getSearchStream() {
+		return this.searchField.asObservable().pipe(
```
Comment (SteveRusin) [inline src/app/home/services/events.service.ts:16]: when you use` .pipe` `.asObservable` becomes redundant.
pipe returns Observable interface

File: src/app/core/services/http-events.service.ts [hunk 2]
```diff
@@ -4,26 +4,26 @@ import { Observable } from 'rxjs';
...
 @Injectable()
 export class HttpServiceEvents{
-    private fireBase = `https://fe-lab-11-team-lemon-project.firebaseio.com/`
-    constructor(private http: HttpClient){ }    
+	private fireBase = `https://fe-lab-11-team-lemon-project.firebaseio.com/`
+	constructor(private http: HttpClient){ }    
 
-    getEvents(request?:string):Observable<any[]> {
-        if(request) {
+	getEvents(request?:string):Observable<any[]> {
```
Comment (SteveRusin) [inline src/app/core/services/http-events.service.ts:10]: you can use default parameter and avoid if else block
Something like this.
Also why you use Observable<any[]> ??
When you use `any` the whole idea of typescript is lost
```ts
	getEvents(request: string = ''): Observable<Event[]> {
			return this.http.get<Event[]>(`${this.fireBase}/events.json/${request}`)
		}

File: src/app/home/components/events/events.component.ts [hunk 3]
```diff
@@ -9,19 +9,33 @@ import { Event } from '../../../core/index'
...
-		this.eventSubscription = this.eventsService.getSearchedEvents().subscribe((value) => {
-            this.events = [];
-			for	(const element in value) {
-				value[element].id = element
-				this.events.push(value[element])
+		this.eventSubscription = combineLatest(this.eventsService.getEventsFromDb(), this.eventsService.getSearchStream())
+		.subscribe(([events, searchStr]) => {
+			this.events = [];
+			for	(const element in events) {
+				events[element].id = element
```
Comment (SteveRusin) [inline src/app/home/components/events/events.component.ts:24]: mutating data ain't a good approach. Can you think of another one?

File: src/app/core/services/http-events.service.ts (hunk 2 above)
Comment (romasaldan) [reply src/app/core/services/http-events.service.ts:10]: This is my mistake. I will fix it in future commit.

**Pull request merged**